porch/controllers/rootsyncsets/api/v1alpha1/rootsyncset_types.go

```diff
@@ -65,6 +65,16 @@ type SecretReference struct {
 type RootSyncSetStatus struct {
 	// Conditions describes the reconciliation state of the object.
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	ClusterRefStatuses []ClusterRefStatus `json:"clusterRefStatuses,omitempty"`
+}
+
+type ClusterRefStatus struct {
+	ApiVersion string `json:"apiVersion,omitempty"`
+	Kind       string `json:"kind,omitempty"`
+	Name       string `json:"name,omitempty"`
+	Namespace  string `json:"namespace,omitempty"`
+	SyncStatus string `json:"syncStatus,omitempty"`
```

Comment:
TBD what status we want to expose, but we have to start somewhere :-)

Also not sure if we should give the target clusters an `id` and key off that here (rather than forcing harder matching), but again ... TBD

Reply:
Yeah, I think at some point we probably want to expose both the synchronization status and reconciliation status from Config Sync, but doing just the sync status at first is easier.

I was considering some options for what to include here to allow us to mach statuses to the clusterRefs from the spec. I ended up duplicating the four identifiers from the clusterRef, because I think this will allow us to remove RootSync from clusters when they are removed from the clusterRef list in the spec. Currently RootSyncSet doesn't handle this, since once the clusterRef is removed, we no longer know which cluster to talk to..